proposed/simple-csharp-programs.md

```diff
@@ -0,0 +1,499 @@
+# Simple C# Programs
+
+This speclet defines "Simple C# Programs" and covers what they can do from the user's standpoint. It does **not** get into detailed design points, a deep dive of a particular mechanisms, etc. The goal is to drive consensus on the user experience and then walk back towards the right design and implementation.
+
+It is also designed to be incremental. Not everything written here must be implemented. For example, we could ship a version without support for packages/frameworks.
+
+## Motivation
+
+C# has lots of great things going for it, but it falls short in a key area: getting started with the least possible friction.
+
+Lots of developers in the world have experience with languages like JavaScript, Python, and Ruby. In these languages, you typically get started in one of these three ways:
+
+1. Launch a REPL and start typing code and getting results
+2. Create a file with the right extension and run the code in that file
+3. Use build tooling to create an application and then run that application
+
+If you look at online guides for the above languages, they tend to start people off with (1) and (2) and rarely with (3).
+
+In the C# world, we start people off with (3) with the base tooling, the .NET CLI.
+
+This is problematic, because these people **do not expect that they need to do (3) to get started**. The tools don't support their instincts and there is very little information online to support them.
+
+Of course, REPLs are another great way to get started. But they are outside the scope of this document.
+
+## What is a Simple C# Program
+
+A Simple C# Program is this:
+
+* A C# file
+* A set of global `using`s inferred by the build environment
+* A tool that can run that file / produce a working program
+* A way to incorporate libraries and frameworks/sdks into that file and run it
+* A way to "graduate" to using a project file
+
+A Simple C# Program is **not** a replacement for project-based C# code or .NET Interactive.
+
+## Simple C# is just C# code
+
+Simple C# Programs are not a dialect of the C# language, like the interactive dialect, CSX. Any code in a Simple C# Program can be built and run in a project-based program.
```

Comment:
Here you say that Simple C# programs will not a dialect of the C# language, but then you propose that the `#r "nuget:PackageId"` syntax be copied from .NET Interactive.

Therefore, are you proposing that directories with a csproj allow individual `.cs` files to also use `#r` to reference packages?

Reply:
Yeah, wondering if it's possible to treat the `#r` syntax as a comment when building with a project file?
We should be able to tell if the build is built with a project/solution file, right?
If yes, think then we could make it different when in different modes